"yarn test" in master branch fails on new installation

## 🐛 Bug Report

I just wanted to contribute. Followed the guide, and got:

```bash
> yarn test
yarn run v1.19.0
$ yarn lint && yarn jest
$ eslint . --cache --report-unused-disable-directives --ext js,jsx,ts,tsx,md

/Users/sergey/Documents/opensource/jest/packages/jest-haste-map/src/lib/FSEventsWatcher.ts
  14:1  error  Unused eslint-disable directive (no problems were reported)

✖ 1 problem (1 error, 0 warnings)
```
To make sure that it is not my fault, I repeated the same with other system, and got:
```bash
> yarn test
yarn run v1.19.0
$ yarn lint && yarn jest
$ eslint . --cache --report-unused-disable-directives --ext js,jsx,ts,tsx,md

/home/plazmer/Documents/jest/packages/jest-worker/src/workers/NodeThreadsWorker.ts
  11:1  error  Unused eslint-disable directive (no problems were reported from 'import/no-unresolved')

/home/plazmer/Documents/jest/packages/jest-worker/src/workers/threadChild.ts
  9:1  error  Unused eslint-disable directive (no problems were reported from 'import/no-unresolved')

✖ 2 problems (2 errors, 0 warnings)
```
So, as we can see, errors occur in different files, but in both cases they are linked to eslint. Maybe need to use specific version? Again - I did not made any changes in both cases, strictly following the given scenario.

## To Reproduce

Steps to reproduce the behavior:
Just follow this guide from scratch - https://github.com/facebook/jest/blob/master/CONTRIBUTING.md

(I did this on two different systems and got eslint errors in both cases)

## Expected behavior

"yarn test" should work without errors.

## envinfo

1) System:
    OS: macOS 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i5-8279U CPU @ 2.40GHz
  Binaries:
    Node: 10.16.3 - /usr/local/bin/node
    Yarn: 1.19.0 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm

2) System:
    OS: Linux 4.15 Ubuntu 16.04.6 LTS (Xenial Xerus)
    CPU: (1) x64 Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz
  Binaries:
    Node: 11.14.0 - /usr/bin/node
    Yarn: 1.19.0 - /usr/bin/yarn
    npm: 6.9.2 - /usr/bin/npm

I can reproduce this error on warm and cold eslint cache:
```
/Users/thymikee/Projects/oss/jest/packages/jest-haste-map/src/lib/FSEventsWatcher.ts
  14:1  error  Unused eslint-disable directive (no problems were reported)
```
Not sure why it differs from the CI.

I'm curious though, how you manage to run yarn 1.19, when we explicitly set `yarn-path` to v1.17 bundled locally in .yarnrc file?

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -21,7 +21,7 @@ jobs:
   lint-and-typecheck:
     working_directory: ~/jest
     docker:
-      - image: circleci/node:10
+      - image: circleci/node:12
     steps:
       - checkout
       - restore-cache: *restore-cache

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -42,7 +42,7 @@ _Before_ submitting a pull request, please make sure the following is done…
     python --version
     ```
 
-1.  Make sure you have a compatible version of `node` installed (As of July 10th 2019, `v10.x` is recommended since there is a known issue with `v12.x` [#8490](https://github.com/facebook/jest/issues/8490)).
+1.  Make sure you have a compatible version of `node` installed (As of October 25th 2019, `v12.x` is recommended).
 
     ```sh
     node -v

--- a/packages/jest-worker/src/workers/NodeThreadsWorker.ts
+++ b/packages/jest-worker/src/workers/NodeThreadsWorker.ts
@@ -7,8 +7,6 @@
 
 import * as path from 'path';
 import {PassThrough} from 'stream';
-// ESLint doesn't know about this experimental module
-// eslint-disable-next-line import/no-unresolved
 import {Worker} from 'worker_threads';
 import mergeStream = require('merge-stream');
 

--- a/packages/jest-worker/src/workers/threadChild.ts
+++ b/packages/jest-worker/src/workers/threadChild.ts
@@ -5,8 +5,6 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-// ESLint doesn't know about this experimental module
-// eslint-disable-next-line import/no-unresolved
 import {isMainThread, parentPort} from 'worker_threads';
 
 import {

